### PR TITLE
Two Tutorials Added: Simulating Illumina Reads and Uploading Data to Cavatica 

### LINKED ISSUE #200: KF download data tutorial -- vidlet suggestions
Comments for Terminal vidlet: wget to download data (From @s-canchi )

The actual content starts between 2-3 seconds and would be good to edit out the initial message about z shell
Actual commands are run within the first minute and 30 seconds and the remainder video is the download process. Maybe edit it out to show the final result ?
Some inconsistency in comment text format with capital and small letters
Empty command lines are distracting
Comments for Terminal vidlet: clean up file names

The actual content starts between 2-3 seconds and would be good to edit out the initial message about z shell
Long pauses between comments and code stretch out the video. Maybe edit it out to show the final result ?
Might work better to use ls -1 to show the filenames since it is a long string
Some inconsistency in comment text format with capital and small letters
Empty command lines are distracting
Observation: Both the screencasts show the cast ID on mouse hover and have the Recorded with asciinema in the bottom left corner.

If you have the original screencast copies, I will be happy to help configure it.

### LINKED ISSUE #262: GWAS general code fix
There's a piece of code that installs ALL the dependencies that somehow got deleted during the last website release. @abhijna will add it in for the December release.

### LINKED ISSUE #267: Fastq read simulation for tutorials and workshops
Find a good fastq read simulator and write a tutorial on how to use it.

### LINKED ISSUE #271: Upload files to Cavatica using the command line uploader thing
This tutorial is the official tutorial but it has got some confusing parts: https://docs.cavatica.org/v1.0/docs/upload-via-the-command-line.

For other tutorials we need to be able to transfer fastq data files from AWS to Cavatica. Here are some steps:

Install java on AWS
Download the uploader onto AWS
Figure out how to use it
Related to simulated data issue #191

### PR DESCRIPTION
**Pull request by abhijna**
## PR Checklist

- [x] Release label added
- [x] PR category label added
- [x] PR mergeable
- [x] Clean build logs
- [x] Linked related issues
- [ ] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description
This is a new tutorial about using InSilicoSeq to simulating fastq data from a user-specified version of the human reference genome. It also includes some QC steps to see how the simulation looks. While this tutorial is good for simulating fastq reads from normal human samples, it cannot simulate mutations or indels.

Need someone to run through everything and make sure 1) that 16GB is enough for it all. 2) that it all works. I haven't done the formatting yet. If there are things you notice and can change easily, please do. If not, leave me a comment and I can fix it when I get to it.

Also added the command line uploader tutorial here. Was going to make a new PR but I pushed changes to this branch by mistake. Sorry :( 
Same deal for this tutorial as well. Check if everything run, look for typos, formatting problems, other ways to improve it. 

**Preview link**
https://cfde-training-and-engagement--263.com.readthedocs.build/en/263/Bioinformatics-Skills/Simulate_Illumina_Reads/

## Review format
Direct editing or comments are ok. 

## Timeline
This is due for December release. Don't care when you give me detailed comments as long as you give me enough time to incorporate your changes. If you can run through the code and let me know if it all installs on your computer in the next two weeks, that would be helpful.

### CODE REVIEW

**Comment by abhijna** [comment]
Maybe Saranya can review the uploader tutorial and Marisa can review the InSilicoSeq? Sorry people. I really should have made two PRs but my brain did a stupid.

**Comment by marisalim** [comment]
@abhijna here you go! got everything to work. I tried with 8Gb of storage, but it wasn't enough (the temp files take up a lot of space i guess). everything worked with the `t2.xlarge` and 16Gb of storage. time=22 minutes for `iss` command, total time to run tutorial is ~1hr, cost=...? (it doesn't show up on my billing dashboard yet) --> $0.18 :)
I tried the whole tutorial on the in-browser terminal. worked fine :)

here are things to edit/add:
### formatting things
- [x]  needs the tutorial landing page info - estimated time, prerequisites, learning objectives. you could put the "Additional Resources" into the "Tutorial Resources" section
- [x] use backticks for AWS instance names and command flags (e.g., `--model`, `--mode basic`)
- [x] update step number format ("Step 1" instead of 1)
- [x] images need captions in the link syntax

### InSilicoSeq tutorial
#### Intro
- [x] maybe give examples of what the answer is (e.g., variants?)/ how 'knowing the answer' helps: "With simulated data, the user knows the final answer"
#### setting up AWS instance
- [x] need info/warning that the t2.xlarge machine is not part of the free tier and will cost some money (add estimated cost and time to run. maybe add to estimated time section of landing page info. speaking of which, how much does the GWAS tutorial cost if you don't have free tier access anymore? we should add a cost estimation if it's not there already)
- [x] AWS tutorial link to the overview page instead of vidlet?
- [x] after choosing t2.xlarge, click on the "4. Add Storage" (e.g., can skip step 3, don't have to click on the 'next:' button)
- [x] say what the 16Gb is for - e.g., the output simulated data files we're generating are x Gb (actually, the outputs are small, it looks like the temp files take up a lot of space)
#### InSilicoSeq
- [x] add link to software page for more info and other options: https://github.com/HadrienG/InSilicoSeq, https://insilicoseq.readthedocs.io/en/latest/. maybe put the additional resources earlier OR link down to them
- [x] when you do the commands below
```
sudo apt update
sudo apt install python3-pip
```
it asks if you want to:
```
1 upgraded, 50 newly installed, 0 to remove and 32 not upgraded.
Need to get 46.8 MB of archives.
After this operation, 200 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```
tell users if they should enter "Y" (same thing for the fastqc installation command)
- [x] there are a bunch of warnings from the insilicoseq installation - maybe add a note that it's ok to ignore these
```
Installing collected packages: pysam, numpy, biopython, scipy, future, joblib, InSilicoSeq
  WARNING: The scripts f2py, f2py3 and f2py3.8 are installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts futurize and pasteurize are installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script iss is installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed InSilicoSeq-1.5.1 biopython-1.78 future-0.18.2 joblib-0.17.0 numpy-1.19.4 pysam-0.16.0.1 scipy-1.5.4
```
- [x] add an `iss --version` check to make sure it's installed properly (also helps if they find they're using a different version from the one in the tutorial)
- [x] say the downloaded human genome file should be 3.1 Gb when uncompressed
- [x] move the "it takes approximately 40 minutes to run" up to the top of the tutorial along with cost estimate
- [x] `--compress true` error (resolved by deleting: this flag works sometimes and sometimes not -_-)
```
iss generate --genomes ../reference_genome/GRCh38_latest_genomic.fna --model hiseq --n_reads 5M --cpus 4 --output my_sim --compress true
usage: iss [subcommand] [options]
InSilicoSeq: error: unrecognized arguments: true
```
- [ ] can you run stuff on AWS and then walk away? (e.g., use screen or tmux) or do you need to keep it open for 40 mins
- outputs:
    - [ ] what do the abundance numbers mean from "my_sim_abundance.txt"?
    - [x] my output files are called "my_sim_R1.fastq" and "my_sim_R2.fastq" (not "my_sim_1.fastq")
    - [ ] could show how to check out the fastq files (`less -S my_sim_R1.fastq"). why are the sequences a mix of lower and upper case letters? (does it matter?)
    - [x] the 5M reads are split between the R1 and R2 (instead of creating 5M each). also I don't have exactly 2.5M per read direction, why? as a simulator, shouldn't it be exact? total, I have 4,999,694 reads total between the R1 and R2 files
- [x] for the for loop command, add that you could do this to simulate data for a bunch of samples otherwise users might not know what to do with that code
- [x] for the warning, can recommend starting out with a very small set of reads to test the code before scaling up
#### Quality control
- [x] edit this, the outputs will not be compressed unless you add a `gzip` step (assuming the `--compress` flag is deleted from `iss` command)
```
cd fastqc <- i think this is supposed to be the fastq directory. there's no fastqc directory
fastqc *.gz <- should we gzip the files? otherwise it will be `fastqc *.fastq`
```
- [x] local transfer - reminder to do this in local terminal and not AWS terminal
- [x] the backticks got included in this code block:
```
scp -i ~/Desktop/amazon.pem ubuntu@ec2-??-???-???-??.us-east-2.compute.amazonaws.com:/home/ubuntu/fastq/\*.html ~/Desktop/fastqc/.
```
- [x] need to type "yes" for local download "Are you sure you want to continue connecting (yes/no/[fingerprint])? yes"
- [x] my fastqc outputs say the read lengths are 126bp not 125bp

#### end of tutorial
- [x] remember to say how to close the instance! and local download the fastq files if they want to keep them.

**Comment by abhijna** [comment]
Thanks, Marisa!! Working on it. A couple of questions (will add more if they come up):

> update step number format ("Step 1" instead of 1)

Do you mean for this section: "Selecting the right AWS instance to run InSilicoSeq"?

--> This is completely up to you how to organize sections into steps. :) I was thinking that the steps under the AWS section could be "### Step 1: Select instance", "### Step 2: Add additional storage", etc. so they show up in the right-side TOC.

> images need captions in the link syntax

Aren't my images captioned? e.g. t2 xlarge

--> the website doesn't display the text in the brackets for some reason so you have to put them in quotes, e.g.,:
t2 xlarge

**Comment by abhijna** [comment]
Documentation about abundance.txt doesn't make sense to me. posted an issue about it: https://github.com/HadrienG/InSilicoSeq/issues/196

**Comment by abhijna** [comment]
There seems to be an open issue about why you get fewer reads than specified in the insilicoseq command

https://github.com/HadrienG/InSilicoSeq/issues/195

**Comment by s-canchi** [comment]
On my radar and will work on it first thing on Monday!

**Review by s-canchi — COMMENTED** [review]
@abhijna - I made some corrections and additions to the data uploader tutorial directly (since I couldn't continue with the tutorial otherwise) but I also have additional comments:

- [x]  In the introduction I am a bit confused by this line: “Cavatica provides….”. Is this tutorial an extension of the listed methods or a better version of the available methods ? Maybe rephrase the sentence to clarify.

- [x] The learning objectives list "Learn to edit metadata of files on Cavatica". But those steps are not present in the current tutorial. Will you be adding more to it ?

- [ ] In Step 2,  for the dowloaded files, would it make sense to add the md5sum check steps ? [Original steps in ANGUS](https://angus.readthedocs.io/en/2019/quality-and-trimming.html)

- [x] In Step 3, could you add some explanation for the different flags used in the tar command.

- [ ] Consider adding some screenshots for the position of username, auth token and the files tab for Cavatica. Alternatively you can also consider adding a screen recording.

**Comment by abhijna** [comment]
Thanks Saranya! I'll work on these in the next couple of days. I thought I had already added the edit metadata sets. I guess not :'(

**Comment by abhijna** [comment]
So thoughts:

1) on md5sum check: for the KF tutorial, the nature of the file we are using in the transfer doesn't really matter. So we don't need to do an md5sum after downloding into AWS. At the Cavatica end, I'm not sure how to do it on their interface :(. Any ideas?
2) >Consider adding some screenshots for the position of username, auth token and the files tab for Cavatica. Alternatively you can also consider adding a screen recording.

I didn't want to add anything here because the" portal set up and permissions tutorial" walks people through finding the authentication token with screenshots. I don't want to have duplicates -- it will be a pain to edit the same thing in two places when the screenshots need to get updated. I've linked that page, though.  Thoughts?

**Comment by abhijna** [comment]
@s-canchi I added the metadata editing section. Could you take a look to make sure it's ok? Should be pretty straightforward I think.

**Comment by s-canchi** [comment]
> on md5sum check: for the KF tutorial, the nature of the file we are using in the transfer doesn't really matter. So we don't need to do an md5sum after downloding into AWS. At the Cavatica end, I'm not sure how to do it on their interface :(. Any ideas?

This is tricky indeed not just for this tutorial but for data in general. So I think of two ways to get this information in Cavatica. 
- Approach one would be verify the md5sum of the files on AWS after download from osf and add the md5sum as a metadata point for the downloaded file to show in the Cavatica interface. This however assumes that the data file integrity is sustained during the upload from AWS to Cavatica. 
- Approach two is to create a CLI in Cavatica using interactive analysis option and to run the md5sum on the downloaded files. I tested this approach and it works. The negative with this approach is that it involves further explanation of steps to set up the interactive analysis. 

I leave the final option to you of whether you want to include these steps or not. But I assume for future tutorials we might want to have some sort of sanity check for data files. 

> I didn't want to add anything here because the" portal set up and permissions tutorial" walks people through finding the authentication token with screenshots. I don't want to have duplicates -- it will be a pain to edit the same thing in two places when the screenshots need to get updated. I've linked that page, though. Thoughts?

That's a valid point. Thinking as a user without much familiarity of the Cavatica interface, I like the visual impact of the screenshots with the circled sections, such that when I go back to the Cavatica interface I can go by visual memory instead of text. Clicking back and forth between two separate tutorials may not be ideal as we want to keep them modular, and hence my suggestion a screen-capture video which can be relatively quick to accomplish.  
I will leave the choice to you about how you would like to incorporate any visual aides. This can also be an enhancement feature that can be tackled next time :)

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    - [Recommended read dept and coverage for NGS applications](https://genohub.com/recommended-sequencing-coverage-by-application/)
+
+    - [What is FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
+
+    - [FastQC video tutorial](https://www.youtube.com/watch?v=bz93ReOv87Y)
+
+
+## Selecting the right AWS instance to run InSilicoSeq
+
+To run [InSilicoSeq](https://github.com/HadrienG/InSilicoSeq), you will need to launch a 64 bit Ubuntu Server 20.04 LTS (HVM), SSD Volume Type instance as described in [the AWS tutorial](Introduction_to_Amazon_Web_Services/introtoaws3.md) with a **few modifications**:
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: let's rephrase this - it's possible to run InSilicoSeq locally on Mac or an HPC. it does not need to be AWS. 

"In this tutorial, we will run InSilicoSeq on a 64 bit Ubuntu Server 20.04 LTS (HVM), SSD Volume Type instance, as described in..."

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+ ### Step 1)
+ You must select the the `t2.xlarge` instance (instead of the default t2.micro instance selection described in the tutorial):
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: format: there are few more places where `t2.micro` and `t2.xlarge` need the backtick formatting

the `###` steps are not showing up in the right-side TOC for some reason. also, nitpicky thing: `### Step 1:` colon instead of `)` please :)

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+[InSilicoSeq](https://insilicoseq.readthedocs.io/en/latest/) is a software that simulates Illumina reads from genomes, and was especially built to simulate reads from metagenomes. InSilicoSeq is written in Python, is fairly well-documented and easily installed via Python’s package manager `pip(3)`.
+
+## Why use simulated data?
+
+There are a myriad of bioinformatics tools currently available to users. Before use, every tool must be tested for general usability and suitability for a given task. Simulated data can be ideal for the purposes of testing bioinformatics tools because the user can model the data set based on their specified parameters. With simulated data, the user knows the final answer, and so they know when the tool doesn't work. For example, when using fastq data that was simulated from a "normal" human reference genome (with no mutation parameters specified), the final answer should be the absence of any detectable variants. If a variant calling pipeline detects variants using this simulated data set, the pipeline doesn't work correctly. Simulated data sets are also great for use in the classroom, especially when protected (human) data can't be distributed.
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: this got a little complicated - how about something like this?

> With so many bioinformatics tools currently available to users, simulated datasets are ideal for quickly generating toy datasets based on real data to test new tools for general usability and user task suitability. For example, a researcher could test a variant calling pipeline on data simulated from the human genome. If the data simulation did not introduce new mutations relative to the original human genome used for simulation, variants should not be detected. Simulated datasets are also useful for teaching, since they can be readily shared unlike protected (human) data."

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+
+InSilicoSeq can be installed using Python's package manager `pip3`. However, `AWS's Ubuntu 20.04` instance does not come pre-installed with `pip3`. Let's begin by updating the instance and installing `pip3`.
+
+=== "AWS Instance Code"
+
+    ```
+    sudo apt update
+    sudo apt install python3-pip
+    ```
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:97]: format: add an empty line between the tabbed box sections, the `sudo` code isn't showing up on the website. same thing for AWS Instance Code boxes below

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+Make a directory called "fastq" in which the simulated fastq files can be saved.
+
+=== "AWS Instance Code"
+
+    ```
+    mkdir fastq
+    cd fastq
+    ```
+
+Finally, run the code to make your simulated fastq file.
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:213]: oops, let's put back the time estimate for this particular step - "The simulation step takes ~30 minutes to complete. (something about leave the terminal open until it's completed...otherwise it may interrupt the run?)"

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    `--model` lets you input an error model file. The default is None. You're using hiseq for a pre-computed error model suitable for simulating data generated from a HiSeq sequencing run. Other available options are NovaSeq and MiSeq. If you do not wish to use a model, use –mode basic. The name of the built-in models is case insensitive.
+
+    `--n_reads` specifies the number of reads to be simulated. The default is 1 million (1M).
+
+    `--cpus` specifies the number of CPUs that should be used. Default is 2.
+
+    `--output` specifies the name of the output files. You're calling it my_sim, but you can change it to whatever suits your needs.
+
+
+If your run is successful, you will see two ".fastq" files, "my_sim_R1.fastq" and "my_sim_R2.fastq", and an abundance file called "my_sim_abundance.txt". The abundance file is more meaningful in the context of simulating metagenomic data. It is a tab-delimited file containing the abundance of each genome supplied to the command. The 5 M reads generated per simulation are split between the R1 and R2 files.
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: oh, so if you simulated data from multiple genomes this would tell you the proportion from each genome in the simulated set? confusing that with 1 input genome, there are so many lines in the output abundance txt file..

i don't know what to do about the imprecise number of simulated reads (since the bug issue on the insilicoseq repo hasn't been answered. i suppose you could add to the thread to say it's a problem for us too). you could add a note that it generates approximately the requested number, but not exactly...?

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+
+## Looping InSilicoSeq
+
+If you want to repeat the simulation multiple times, modify the following code block by changing the `n` to an integer corresponding to the number of time you wish to run the simulation. Your output files will be named 1_my_sim_R1 and 1_my_sim_R2, 2_my_sim_R1 and 2_my_sim_R2, ..., n_my_sim_R1 and n_my_sim_R2.
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: minor edits: 

If you want to simulate data for multiple samples, modify the following code block by changing the `n` to the number of samples you want to simulate. Your output files will be named:

-  "1_my_sim_R1.fastq" and "1_my_sim_R2.fastq"
-  "2_my_sim_R1.fastq" and "2_my_sim_R2.fastq"
- ...
-  "n_my_sim_R1.fastq" and "n_my_sim_R2.fastq"

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+=== "AWS Instance Code"
+
+    ```
+    for i in {1..n};
+    do iss generate --genomes ../reference_genome/GRCh38_latest_genomic.fna --model hiseq --n_reads 5M --cpus 4 --output ${i}_my_sim;
+    done
+    ```
+
+!!! Warning
+    Depending on how many times you run the simulation and/or how many reads you simulate, the code may take a really long time to complete! You can start out by running a small set of reads (2-3) to test the code before scaling up.
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: "small set of reads (`--n_reads 1M`)" something like this to show which flag to edit

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+
+Here is an image of the top of an example FastQC report.
+
+![](../images/Simulated_Data_Fastqc.png "FastQC report")
+
+
+### Interpretation
+
+**Watch this [YouTube video](https://www.youtube.com/watch?v=bz93ReOv87Y) on how to interpret FastQC results**:
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: Add in a reference that this video was made by the Babraham Bioinformatics Institute (the developers of FastQC).

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+
+[![FastQC video tutorial](http://img.youtube.com/vi/bz93ReOv87Y/0.jpg)](https://www.youtube.com/watch?v=bz93ReOv87Y "FastQC Interpretation")
+
+All simulated reads are 126 bp long :)
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: edit suggestion: "From the FastQC report, we see that the overall read quality scores are good and read length is 126bp. These simulated reads are now ready to be used for analysis!"

File: docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+
+!!! note "AWS Instance Termination"
+
+    - To avoid unnecessary charges, remember to [terminate your AWS instance](Introduction_to_Amazon_Web_Services/introtoaws4.md) once you are done using it.
+
+    - It is important to remember that terminating the instance causes all the data in that instance to be lost forever. If you wish to keep the fastq files, please remember to transfer them to your local machine or analysis platform. Check out our [tutorial on transferring files from AWS to Cavatica](Kids-First/Upload_Data.md).
```
Comment (marisalim) [inline docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md]: hm, is there a reason to move these fastq files to Cavatica? vs. the more general instructions for downloading to local computer? (also because not everyone doing this tutorial will have the login set up to Cavatica)

**Review by marisalim — COMMENTED** [review]
@abhijna looking good! here are a few more edits. it would be good to get 1 more person to run through the tutorial!

**Comment by abhijna** [comment]
Ok, I made all the changes. Thank you :)

 You're right, I think we need one more person to run through the tutorial to see if it all makes sense and runs correctly. Maybe @josesanchez1815 or @jeremywalter

**Comment by josesanchez1815** [comment]
I'll take a look

**Comment by jeremywalter** [comment]
TOC in Illumina looks wonky because it just says
Step 1: 
Step 2: 
Step 3:
Step 4: 

How about -->
Step 1: Select Instance
Step 2: Add Storage
Step 3: Review and Launch
Step 4: Access the Instance

(also... I'm doing this ^^^)

File: docs/Bioinformatics-Skills/Kids-First/Upload_Data.md [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+
+- Next, select the files whose metadata you wish to edit by checking the box next to the file name.
+
+![](images/Select_Samples.png "Select the Files")
+
+- Click the "Edit Metadata" button
+
+![](images/Edit_Metadata_Button.png "Edit Metadata Button")
+
+- You should see a pop-up window on the right side of the screen
```
Comment (s-canchi) [inline docs/Bioinformatics-Skills/Kids-First/Upload_Data.md]: It seems that Cavatica has set metadata fields that you can edit. Is there a way to add new metadata terms if not listed in the options ?

**Comment by jeremywalter** [comment]
Thoughts and observations (Illumina Reads): 

1) I did not run through the tutorial but the bottom steps seems like they might trip me up  -- I'm betting the transfer to local computer part might be confusing for others (as it would be potentially for me)... paths and different terminals are confusing! I'm not sure how to fix that without a screencast. When I clicked on the link (path to amazon.pem) I felt more confused.

2) At the end it warns users to turn off their instance. I suggest putting part or all of this in the intro. Why? Because when I ran the AWS long ago I got e-mails about my usage for weeks and weeks and finally turned it off after the lesson we just had. Why? Maybe I didn't finish the AWS tutorial... sometimes people don't work their way to the end. Since I was using free-tier nothing happened to me, but it's a good warning to preface any cloud lesson with.. 'don't forget to turn it off when you're done or $$$' 

Note: I'm just here looking for what's wrong... but I love all the 'enter this code' and 'here's the output' steps... :)

File: docs/Bioinformatics-Skills/Kids-First/Upload_Data.md (hunk 12 above)
Comment (abhijna) [reply docs/Bioinformatics-Skills/Kids-First/Upload_Data.md]: I'm 98% sure that the answer is no, but I will double check.

**Comment by abhijna** [comment]
@jeremywalter thank you! this is helpful. I will rework it.

**Comment by josesanchez1815** [comment]
This is form the Simulating Illumina reads

HI! I think the only comments I have for this tutorial is I got confused about when to use the fastqc command? 
It was an unsuccessful run, because I wasn't sure which folder to be in.

So, when after the command 
`sudo apt install fastqc` 
I enter `cd fastq`

and I created a new empty folder with no fastqc files and tried to run fastqc there. Obviously, it didn't do anything because I did not have any files.

So I think maybe omit `cd fast q` because we are already inside the fastq folder at this point.

**Comment by jeremywalter** [comment]
image
This section from uploading to Cavatica is curiously absent of explanation (compared to the rest of the lesson). What's tar? what's the -C flag and the mysterious ~?

File: docs/Bioinformatics-Skills/Kids-First/Upload_Data.md (hunk 12 above)
Comment (abhijna) [reply docs/Bioinformatics-Skills/Kids-First/Upload_Data.md]: nevermind there is a roundabout way of doing this. significant addition coming up.

Comment (marisalim) [reply docs/Bioinformatics-Skills/Kids-First/Upload_Data.md]: @abhijna i can test this tutorial when you're done adding this metadata part

Comment (abhijna) [reply docs/Bioinformatics-Skills/Kids-First/Upload_Data.md]: postponed to the next release. it's not working right now. will fuss with it some at a later time.

**Comment by abhijna** [comment]
I made all the changes. Put two of the comments (about md5sum and screencasts) into issues for the next release.

**Comment by jeremywalter** [comment]
RE: minor edits commit -- I already did that? could create a conflict? leaving this here for reference in case it does (how did that happen?)

**Comment by marisalim** [comment]
@abhijna for the Cavatica uploader tutorial:
- all the steps worked!
- minor: the cavatica uploader is a ".tgz" file, but the explanation of the `tar` command talks about "tar.gz" extension. are they the same? ---> yes it is the same. I changed it to say .tgz :) 
- this can be an edit for next iteration, but i think it would be best to stick to 1 example dataset - show the yeast example files in the Cavatica screenshots instead of the sim data ones. ---> yeah, i'd have to redo all the screenshots. so this can be a thing for next time. 
- add a note/link to close down the AWS instance at the end of the tutorial. ---> done

**Comment by abhijna** [comment]
Done :)

**Review by marisalim — APPROVED** [review]
looks like all the changes i requested got added! good to go
